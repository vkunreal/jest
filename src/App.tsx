import React, { useRef, useState } from "react";

const App = () => {
  const [value, setValue] = useState("");
  const [login, setLogin] = useState("");
  const inputRef = useRef<HTMLInputElement | null>(null);

  const handleChange = () => {
    setValue(inputRef.current!.value);
  };

  const handleKeyUp = (e: React.KeyboardEvent<HTMLInputElement>) => {
    if (e.code == "Enter") handleClick();
  };

  const handleClick = () => {
    setLogin(value);
    setValue("");
  };

  return (
    <div data-testid="appComp">
      <h1>Enter the data:</h1>

      <input
        type="text"
        value={value}
        onChange={handleChange}
        onKeyUp={handleKeyUp}
        ref={inputRef}
        data-testid="inputElem"
      />
      <button onClick={handleClick} data-testid="btnElem">
        Send
      </button>

      {login && (
        <div>
          <p data-testid="loginField">{login}</p>
        </div>
      )}
    </div>
  );
};

export default App;
