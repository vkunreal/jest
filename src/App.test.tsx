import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import App from "./App";

describe("App.tsx", () => {
  test("Renders heading", () => {
    render(<App />);
    const linkElement = screen.getByText(/enter the data:/i);
    expect(linkElement).toBeInTheDocument();
  });

  test("Renders login", () => {
    render(<App />);

    const input = screen.getByTestId("inputElem");
    const btn = screen.getByTestId("btnElem");
    const testValue = "test-login";

    userEvent.type(input, testValue);
    userEvent.click(btn);

    expect(screen.queryByTestId("loginField")).toContainHTML(testValue);
  });

  test("Snapshot", () => {
    render(<App />);

    const appComp = screen.getByTestId("appComp");

    expect(appComp).toMatchSnapshot();
  });
});
