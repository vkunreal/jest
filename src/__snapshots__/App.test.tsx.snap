// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.tsx Snapshot 1`] = `
<div
  data-testid="appComp"
>
  <h1>
    Enter the data:
  </h1>
  <input
    data-testid="inputElem"
    type="text"
    value=""
  />
  <button
    data-testid="btnElem"
  >
    Send
  </button>
</div>
`;
